gdpr/models.py

```diff
@@ -78,6 +81,25 @@ def exists_valid_consent(self, purpose_slug, source_object):
 
 class LegalReasonQuerySet(models.QuerySet):
 
+    def filter_expired_retaining_data_in_last_days(self, days=None):
+        """
+        Filters all Legal Reason that retain data and that expired in last days
+
+        Args:
+            days: Number of days in the past. If not provided, all Legal Reasons retaining data which expired in the
+            past will be returned.
+        """
+        purpose_slugs_retaining_data = [slug for slug, cls in purposes_map.items() if cls.is_retaining_data]
+
+        filter_keys = {
+            'expires_at__lt': timezone.now(),
+        } if days is None else {
+            'expires_at__gt': timezone.now() - relativedelta(days=days),
+            'expires_at__lt': timezone.now()
+        }
+
+        return self.filter(Q(**filter_keys), Q(purpose_slug__in=purpose_slugs_retaining_data))
```

Comment:
ty q tu nepotřebuješ ne?

Reply:
Ano, díky.Pozůstatek po předchozích složitějších pokusech;-(